### PR TITLE
Document OpenSearch known issue for 2024.1 upgrade

### PR DESCRIPTION
**Pull request by MoteHue**

### CODE REVIEW

File: doc/source/operations/upgrading-openstack.rst [hunk 1]
```diff
@@ -137,6 +137,23 @@ Known issues
...
+
+  ..code-block:: console
+
+   curl -X PUT "https://<kolla-vip>:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
+   {
+      "transient" : {
+         "cluster.routing.allocation.enable" : "all"
+      }
+   }
+   '
```
Comment (Alex-Welsh) [inline doc/source/operations/upgrading-openstack.rst]: I think your spacing is off here. Needs to be in line with the start of `code-block` e.g.:
```
  ..code-block
    text
```

Comment (MoteHue) [reply doc/source/operations/upgrading-openstack.rst]: Done

File: doc/source/operations/upgrading-openstack.rst [hunk 2]
```diff
@@ -137,6 +137,23 @@ Known issues
   applying package updates. This will happen automatically as a post hook when
   running the ``kayobe overcloud host package update`` command.
 
+* After upgrading OpenSearch to the latest 2023.1 container image, we have seen
+  cluster routing allocation be disabled on some systems. See bug for details:
+  https://bugs.launchpad.net/kolla-ansible/+bug/2085943.
```
Comment (Alex-Welsh) [inline doc/source/operations/upgrading-openstack.rst:142]: nit: This should probably be a real rst link

**Review by Alex-Welsh — DISMISSED** [review]

File: doc/source/operations/upgrading-openstack.rst (hunk 2 above)
Comment (MoteHue) [reply doc/source/operations/upgrading-openstack.rst:142]: I don't think it needs to be, I'm not hyperlinking a word, just pointing to the url.

**Review by Alex-Welsh — APPROVED** [review]

**Pull request merged**